index.js

```diff
@@ -158,9 +177,14 @@ function parse (opts, cb) {
     if (!fs.existsSync(file) || fs.statSync(file).isDirectory()) {
       var filename = './' + path.basename(file)
       debug('resolve', [path.dirname(file), filename])
-      file = resolve.sync(filename, { basedir: path.dirname(file), extensions: extensions })
+      file = resolve.sync(filename, { basedir: path.dirname(file), extensions: Object.keys(extensions) })
     }
 
+    var ext = path.extname(file)
+    var detective = extensions[ext]
+
+    if (typeof detective !== 'function') return callback(new Error('Detective function missing for "' + ext + '"'))
```

Comment:
> Will it be triggered whenever an unknown file extension is encountered?

It shouldn't encounter an unknown file extension, since to get into this function the file must first exist using `resolve`. This means that it exists the same as the current behaviour, just with a slightly different message to indicate the extension.